python/tskit/tables.py

```diff
@@ -1666,22 +1666,33 @@ def simplify(
     def map_ancestors(self, samples, ancestors):
         """
         Returns an :class:`EdgeTable` instance describing how ``samples`` are descended
-        from ``ancestors`` (and how ``ancestors`` are descended from each other).
+        from ``ancestors``.
         Each row ``parent, child, left, right`` in the output table indicates that
-        ``child`` (which is either in ``samples`` or in ``ancestors``) has inherited the
-        segment ``[left, right)`` from ``parent`` (which is in ``ancestors``) more
-        recently than from any other node in ``ancestors``. Every segment of the
-        genome on which each of ``samples`` has inherited from any of ``ancestors``
-        is described in this way, as well as any segment of the genomes of ``ancestors``
-        that has been inherited by any of ``samples``.  The node IDs in ``parent`` and
-        ``child`` refer to the node table in the original tree sequence. In other words,
-        the tree constructed at a position with the output edge table has the property
-        that the parent of any node in ``samples``, if there is one, is the most recent
-        ancestor of that node at that position out of all possible ancestors in
-        ``ancestors``, and the same is true for any node in ``ancestors`` that is
+        ``child`` has inherited the segment ``[left, right)`` from ``parent`` more
+        recently than from any other node in these lists.
+        The following table shows which ``parent->child`` pairs will be shown in the output
+        of ``map_ancestors``:
+
+        |Type of relationship      | Shown in output of ``map_ancestors``           |
+        | ------------------------ | ---------------------------------------------- |
+        |``ancestor->sample``      | Always                                         |
+        |``ancestor1->ancestor2``  | Only if ``ancestor2`` has a relevant descendant|
+        |``sample1->sample2``      | Always                                         |
+        |``sample->ancestor``      | Only if ``ancestor`` has a relevant descendant |
+
+        In other words, the tree constructed at a position with the output edge table has
+        the property that the parent of any node in ``samples``, if there is one, is the
+        most recent ancestor of that node at that position out of all possible ancestors
+        in ``ancestors``, and the same is true for any node in ``ancestors`` that is
         ancestral to any of ``samples``.
```

Comment:
I think that the table contradicts the text here? The text says that the parent of any sample is one of `ancestors`; but the table says that one sample can be a parent to another sample? [The tests](https://github.com/tskit-dev/tskit/blob/7f24d6adcbdfe8de5510c702895bfd55293da60f/python/tests/test_topology.py#L3695) suggest that *only* `ancestors` can appear in the `parents` column; so shouldn't the 3rd and 4th lines of the table be "Never"?

Reply:
(sorry for the delay on this - have been busy with other things)

> The text says that the parent of any sample is one of ancestors; but the table says that one sample can be a parent to another sample?

I've changed the first sentence back to something quite vague now - I'm not sure that there is a clear way to describe everything that this function will and won't do in particular cases in a single, easy-to-parse sentence, so I'm hoping users will look at the table for this now.

> The tests suggest that only ancestors can appear in the parents column; 

Ah, thanks for pointing this out -- so this line of code is in the `verify` function, which was initially only being used to test situations where the samples were in the final generation of the simulation (ie. the last two rows of the table weren't being tested here). I have now changed the function, and expanded the tests using it so that all rows of the table are being tested.

> shouldn't the 3rd and 4th lines of the table be "Never"?

nup, see above. Btw, there are 'bespoke' tests covering the behaviour described in each line of the table.

1. [here](https://github.com/tskit-dev/tskit/blob/e55fe76a113ff13914bfbea900e889a862bec20b/python/tests/test_topology.py#L3672) (for example)
2. [here](https://github.com/tskit-dev/tskit/pull/266/files#diff-5941ccebfc6d10311b8d79e8bc986cffR3632)
3. [here](https://github.com/tskit-dev/tskit/pull/266/files#diff-5941ccebfc6d10311b8d79e8bc986cffR3652)
4. [here](https://github.com/tskit-dev/tskit/pull/266/files#diff-5941ccebfc6d10311b8d79e8bc986cffR3642)